source/vstd/hash_map.rs

```diff
@@ -110,6 +110,14 @@ impl<Key, Value> HashMapWithView<Key, Value> where Key: View + Eq + Hash {
     {
         self.m.clear()
     }
+
+    #[verifier::external_body]
+    pub fn extend(&mut self, other: Self)
+        ensures
+            self@ == old(self)@.union_prefer_right(other@),
+    {
+        self.m.extend(other.m)
```

Comment:
Jay Lorch points out that there is no `<HashMap as _>::extend` method on `std::collection::HashMap`. I believe this works because it desugars to `<HashMap as std::tier::Extend>::extend(self.m, other.m)` where the signature of `extend` is `fn [extend](https://doc.rust-lang.org/std/iter/trait.Extend.html#tymethod.extend)<T>(&mut self, iter: T)
       where T: [IntoIterator](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html)<Item = A>;` and `std::collection::HashMap` implements `IntoIterator<Item = (K, V)>`.
So, strictly speaking, this is a different API than the one offered by the regular `HashMap::extend`, so we may want to name it `union_prefer_right`, instead.

Reply:
Updated